NBitcoin/Protocol/Node.cs

```diff
@@ -295,10 +297,15 @@ private void Cleanup(Exception unhandledException)
 				if(Node.State != NodeState.Failed)
 					Node.State = NodeState.Offline;
 
-				_Cancel.Cancel();
+				if(_Cancel.IsCancellationRequested == false)

```

Comment:
I don't understand the reasons for the two if conditions can you explain?

Reply:
I had some unit tests failing cause of that CancellationTokenSource either being disposed or already cancelled by the time the EndListen method gets called. It can also be already cancelled if the node shuts down or cancellation was requested by the user.

There are some race conditions whereby the node's disconnect methods gets called before the listener and receiving thread have properly finished executing. Which means the ManualResetEvent or CancellationTokenSource   in NodeConnection can be disposed or closed or already by the time it gets here in the code.

Reply:
I believe `_Cancel.Cancel();` is immutable right? do you get errors if you call cancel on a cancelled token?